Build-failures like

```
packunused.hs:15:48: error:
    Module ‘Distribution.Package’ does not export ‘unUnitId'
```

means that the selected lib:Cabal version doesn't provide the expected API. I see `build-depends` still allows `Cabal-1.24.*`; should this rather be `build-depends: Cabal == 2.0.*` ?

--- a/packunused.cabal
+++ b/packunused.cabal
@@ -77,7 +77,7 @@ executable packunused
   other-extensions:    CPP, RecordWildCards
   ghc-options:         -Wall -fwarn-tabs -fno-warn-unused-do-bind
   build-depends: base                 >= 4.5    && < 4.11
-               , Cabal                >= 1.24   && < 2.1
+               , Cabal                >= 2.0    && < 2.1
                , optparse-applicative >= 0.8    && < 0.15
                , directory            >= 1.1    && < 1.4
                , filepath             >= 1.3    && < 1.5

--- a/packunused.hs
+++ b/packunused.hs
@@ -1,4 +1,4 @@
-{-# LANGUAGE CPP, RecordWildCards, LambdaCase #-}
+{-# LANGUAGE CPP, RecordWildCards #-}
 
 module Main where
 
@@ -145,7 +145,7 @@ main = do
             when res $ putStrLn "*WARNING* outdated config-data -- please re-configure"
 
     let cbo = map
-          (\case
+          (\x -> case x of
               LibComponentLocalBuildInfo {componentLocalName = n} -> n
               FLibComponentLocalBuildInfo {componentLocalName = n} -> n
               ExeComponentLocalBuildInfo {componentLocalName = n} -> n

